src/RenderWebGL.js

```diff
@@ -737,6 +737,40 @@ class RenderWebGL extends EventEmitter {
     }
 
     /**
+     * Update the position object's x & y members to keep the drawable fenced in view.
+     * @param {int} drawableID - The ID of the Drawable to update.
+     * @param {object} position - An object with x & y members to be fenced
+     *                          - These values are updated in the object to become fenced
+     */
+    getFencedPositionOfDrawable (drawableID, position) {
+        const drawable = this._allDrawables[drawableID];
+        if (!drawable) {
+            // TODO: fix whatever's wrong in the VM which causes this, then add a warning or throw here.
+            // Right now this happens so much on some projects that a warning or exception here can hang the browser.
+            return;
+        }
+        const dx = position.x - drawable._position[0];
+        const dy = position.y - drawable._position[1];
+
+        const aabb = drawable.getFastBounds();
+
+        // This is my best guess at the fencing in Scratch 2,
+        // but I suspect it may need further work to be precisely the same?
+        const sx = this._xRight - Math.min(15, Math.floor((aabb.right - aabb.left) / 2));
```

Comment:
Could you pull the 15 out into a `const` and give it a name? I think it's fine for it to live in this function, but I think giving it a name would make this code more readable to a new Scratch contributor.